lib/register.coffee

```diff
@@ -15,37 +15,38 @@ limitations under the License.
 ###
 
 Promise = require('bluebird')
+postAsync = Promise.promisify(require('request').post, multiArgs: true)
 _ = require('lodash')
 
+randomHexChar = ->
+	return _.random(15).toString(16)
```

Comment:
This was using `Math.random` before, check https://github.com/resin-io-modules/resin-register-device/pull/22/files/6dcf9f3a78e1544b2cec84ee792282b166746a01#diff-d23d2f1f8dd77d2219c945046bbaf969L46 - all I did was to make use of lodash functions to reduce the amount of code required to do the same thing  
As for why, I remember a big convo with @petrosagg involved so he can probably elaborate